app/models/manageiq/providers/cloud_manager.rb

```diff
@@ -73,12 +73,16 @@ def sync_tenants
 
       CloudTenant.with_ext_management_system(id).walk_tree do |cloud_tenant, _|
         cloud_tenant_description = cloud_tenant.description.blank? ? cloud_tenant.name : cloud_tenant.description
-        tenant_params = {:name => cloud_tenant.name, :description => cloud_tenant_description}
+        tenant_params = {:name => cloud_tenant.name, :description => cloud_tenant_description, :source => cloud_tenant}
+
+        tenant_parent = cloud_tenant.parent.try(:source_tenant) || source_tenant
```

Comment:
I'm not too familiar with cloud_tenants but do they also have a parent?

Reply:
Yes, cloud_tenants have parent tenants and sub tenants just like ours.